File: solutions/java/acronym/1/src/main/java/Acronym.java
class Acronym {

    private final String phrase;

    Acronym(String phrase) {
        this.phrase = phrase;
    }

    String get() {
        String text = phrase.replace("-", " ");
        text = text.replaceAll("[^a-zA-Z0-9 ]", "");

        StringBuilder acronym = new StringBuilder();

        for (String word : text.split("\\s+")) {
            if (!word.isEmpty()) {
                acronym.append(Character.toUpperCase(word.charAt(0)));
            }
        }

        return acronym.toString();
    }
}

File: solutions/python/acronym/1/acronym.py
import re

def abbreviate(phrase):
    # Replace hyphens with spaces (word separators)
    phrase = phrase.replace("-", " ")

    # Remove all punctuation except letters, numbers, and spaces
    phrase = re.sub(r"[^A-Za-z0-9 ]", "", phrase)

    # Split into words
    words = phrase.split()

    # Take the first letter of each word and uppercase it
    acronym = "".join(word[0].upper() for word in words)

    return acronym
